syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionIrcConnectorJSONLoadingTest.java

```diff
@@ -0,0 +1,180 @@
+
+/**
+ * @author lhein
+ */
+public class SyndesisExtensionIrcConnectorJSONLoadingTest {
+
+	private SyndesisExtension extension;
+
+	@Before
+	public void onStart() throws IOException {
+		File jsonFile = new File("testdata/irc-connector.json");
+		this.extension = SyndesisExtension.getJSONFactoryInstance(new FileInputStream(jsonFile));
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getExtensionId()}.
+	 */
+	@Test
+	public void testGetExtensionId() {
+		assertEquals("extensionId value is not matching", "io.syndesis.extensions:syndesis-connector-irc", extension.getExtensionId());
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getVersion()}.
+	 */
+	@Test
+	public void testGetVersion() {
+		assertEquals("version value is not matching", "1.0.0", extension.getVersion());
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getName()}.
+	 */
+	@Test
+	public void testGetName() {
+		assertEquals("name value is not matching", "IRC", extension.getName());
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getDescription()}.
+	 */
+	@Test
+	public void testGetDescription() {
+		assertEquals("description value is not matching", "IRC Extension Connector for Syndesis", extension.getDescription());
+	}
+	
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getDependencies()}.
+	 */
+	@Test
+	public void testGetDependencies() {
+		assertThat(extension.getDependencies()).filteredOn("id", "io.syndesis.integration:integration-runtime:jar:1.3-SNAPSHOT").isNotNull(); 
+		assertThat(extension.getDependencies()).filteredOn("id", "org.apache.camel:camel-core:jar:2.20.1").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.apache.camel:camel-irc:jar:2.20.1").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.apache.camel:camel-spring-boot:jar:2.20.1").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.springframework.boot:spring-boot-autoconfigure:jar:1.5.8.RELEASE").isNotNull();
+		assertThat(extension.getDependencies()).filteredOn("id", "org.springframework.boot:spring-boot-loader:jar:1.5.8.RELEASE").isNotNull();
+	}
+
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getExtensionType()}.
+	 */
+	@Test
+	public void testGetExtensionType() {
+		assertEquals("extension type value is not matching", "Connectors", extension.getExtensionType());
+	}
+		
+	/**
+	 * Test method for {@link org.fusesource.ide.syndesis.extensions.core.model.SyndesisExtension#getActions()}.
+	 */
+	@Test
+	public void testGetActions() {
+		assertTrue("actions are not correct", extension.getActions() != null && !extension.getActions().isEmpty() && extension.getActions().size() == 1);
```

Comment:
I think that using `assertThat(extension.getActions()).hasSize(1);` is providing the same level of check

Reply:
true